test/pkg/environment/common/setup.go

```diff
@@ -75,33 +75,52 @@ func (env *Environment) BeforeEach() {
 }
 
 func (env *Environment) ExpectCleanCluster() {
+	env.ExpectSystemNodesTainted()
+	env.ExpectNoPodsInDefaultNamespace()
+	env.ExpectNoProvisionablePods()
+
+	// This assumes the cluster is created without managed pools
+	env.ExpectNoNodePoolOrAKSNodeClass()
```

Comment:
Note that technically this belongs into Azure-specific, rather than common environment - but this is a pre-existing imperfection

Reply:
Yeah I was thinking the same thing.